### DISCUSSION TITLE
Validation using @sinclair/typebox library as per recommendation, not working?

### DISCUSSION THREAD

**Discussion by GlitchTech** [cyrilletuzi/angular-async-local-storage#1010] [Q&A] [answered]
I am building out a new validation for an object type and Angular is throwing errors. I'm trying to figure out why it is throwing this complaint in the code and at runtime. According to the [validation documentation](https://github.com/cyrilletuzi/angular-async-local-storage/blob/main/docs/VALIDATION.md#how-to-validate-advanced-data), it should work? Am I missing something? I checked and I'm using 17.0.0.

**Error**
`Argument of type 'TObject<{ events: TArray<id: TString;>; schedule: TString; }>' is not assignable to parameter of type 'JSONSchema'.`

**Call**
`this.storage.get<EventList>(KEY_EventList, SCHEMA_EventList);`

**Definitions**
`const SCHEMA_EventList = Type.Object({
    events: Type.Array(
        id: Type.String()
    ),
    schedule: Type.String(),
});`

`type EventList = Static<typeof SCHEMA_EventList>;`

**Comment by cyrilletuzi** [comment, marked as answer]
Hello,

There is a syntax error in your schema, remove `id: ` and it is OK:

```ts
const SCHEMA_EventList = Type.Object({ events: Type.Array( Type.String() ), schedule: Type.String(), });
```

**Comment by GlitchTech** [comment]
Ballocks. Was that it? Thanks.
